### PR TITLE
Add sample CLI dockerfiles

### PR DESCRIPTION
**Pull request by awharn**
Adds 4 sample dockerfiles for the CLI with README documentation and workflows.

### CODE REVIEW

**Review by gejohnston — APPROVED** [review]
LGTM

**Review by zFernand0 — APPROVED** [review]
cool techniques overall! 🥳 
LGTM! 😋

**Comment by MarkAckert** [comment]
Looks great! One question on - `Modifies OpenSSL to allow for backwards compatibility with TLS 1.1 and TLS 1.0 servers.` 

Do we know about common TLS1.0/1.1 cases we anticipate running into? Since they're considered insecure, I would argue its better to default 1.2 and include instructions on how to downgrade TLS or create a separate line of images with the downgraded TLS support.

Either way, I support merging this PR - the 1.2 question can be an issue for future consideration.

**Review by MarkAckert — APPROVED** [review]

**Comment by awharn** [comment]
> Looks great! One question on - `Modifies OpenSSL to allow for backwards compatibility with TLS 1.1 and TLS 1.0 servers.`
> 
> Do we know about common TLS1.0/1.1 cases we anticipate running into? Since they're considered insecure, I would argue its better to default 1.2 and include instructions on how to downgrade TLS or create a separate line of images with the downgraded TLS support.
> 
> Either way, I support merging this PR - the 1.2 question can be an issue for future consideration.

I can remove this, we have been doing this in some of the images I was basing this on, but I think it was more common of an issue when TLS 1.3 was just rolling out. I'll leave the line that implements the behavior in but commented out.

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @awharn!

Maybe in the future we can create a reusable workflow - seems like the Docker build/push workflows have a lot in common 🙂

**Comment by awharn** [comment]
> LGTM, thanks @awharn!
> 
> Maybe in the future we can create a reusable workflow - seems like the Docker build/push workflows have a lot in common 🙂

I might look into that today, as the last part of my innovation sprint :)

File: cli/alpine-basic/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Zowe CLI Docker Container
+
+This repository contains the files required to create a Zowe CLI docker container with root access on Alpine
+Does not install the DB2 plug-in
```
Comment (t1m0thyj) [inline cli/alpine-basic/README.md]: Should we mention the reason why? I assume because it's not supported on some architectures such as ARM 😋

Comment (awharn) [reply cli/alpine-basic/README.md]: Probably!  Will do that, it is because DB2 requires GLIBC libraries to build.

**Comment by awharn** [comment]
@MarkAckert I want to make sure that the licenses in the files look good before I merge this. Some are after the shebang line because otherwise files such as the docker entrypoint error out on the created image. Others don't have it because the files are just being `cat`ed into other files.

**Pull request merged**